From 989045c517748f128c6a8fd0300c1adb1b021d7d Mon Sep 17 00:00:00 2001
From: Andrew Burgess <andrew.burgess@embecosm.com>
Date: Tue, 6 Mar 2018 16:27:13 +0000
Subject: [PATCH 8/9] gdb: Add riscv to list of architectures with a
 save_reggroup

The regcache cooked_read test needs to know which architectures have a
save_reggroup, riscv does and needs adding to the list.

gdb/ChangeLog:

	* regcache.c (cooked_read_test): Add riscv to the list of
	architectures that have a save_reggroup.
---
 gdb/ChangeLog  | 5 +++++
 gdb/regcache.c | 3 ++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/gdb/ChangeLog b/gdb/ChangeLog
index d384e07c5b..166b69b884 100644
--- a/gdb/ChangeLog
+++ b/gdb/ChangeLog
@@ -1,3 +1,8 @@
+2018-08-23  Andrew Burgess  <andrew.burgess@embecosm.com>
+
+	* regcache.c (cooked_read_test): Add riscv to the list of
+	architectures that have a save_reggroup.
+
 2018-08-23  Andrew Burgess  <andrew.burgess@embecosm.com>
 
 	* riscv-tdep.c (riscv_register_name): Remove target description
diff --git a/gdb/regcache.c b/gdb/regcache.c
index fb6a904dec..b6e6eda7ee 100644
--- a/gdb/regcache.c
+++ b/gdb/regcache.c
@@ -1895,7 +1895,8 @@ cooked_read_test (struct gdbarch *gdbarch)
 	      || bfd_arch == bfd_arch_msp430 || bfd_arch == bfd_arch_mep
 	      || bfd_arch == bfd_arch_mips || bfd_arch == bfd_arch_v850_rh850
 	      || bfd_arch == bfd_arch_tic6x || bfd_arch == bfd_arch_mn10300
-	      || bfd_arch == bfd_arch_rl78 || bfd_arch == bfd_arch_score)
+	      || bfd_arch == bfd_arch_rl78 || bfd_arch == bfd_arch_score
+	      || bfd_arch == bfd_arch_riscv)
 	    {
 	      /* Raw registers.  If raw registers are not in save_reggroup,
 		 their status are unknown.  */
-- 
2.18.0

